src/preamble.js

```diff
@@ -46,12 +46,33 @@ function getSafeHeapType(bytes, isFloat) {
 var SAFE_HEAP_COUNTER = 0;
 #endif
 
+var safe_heap_warn_set = new Set();
+
 function SAFE_HEAP_STORE(dest, value, bytes, isFloat) {
 #if SAFE_HEAP_LOG
   Module.print('SAFE_HEAP store: ' + [dest, value, bytes, isFloat, SAFE_HEAP_COUNTER++]);
 #endif
   if (dest <= 0) abort('segmentation fault storing ' + bytes + ' bytes to address ' + dest);
-  if (dest % bytes !== 0) abort('alignment error storing to address ' + dest + ', which was expected to be aligned to a multiple of ' + bytes);
+  if (dest % bytes !== 0)
```

Comment:
Yeah, I think improving safe heap can be orthogonal to this. It would make sense for wasm too, but there are some questions as to how to do it. Might make sense to open a separate issue.